packages/grafana-ui/src/components/DateTimePickers/DatePickerWithInput/DatePickerWithInput.tsx

```diff
@@ -27,78 +27,76 @@ export interface DatePickerWithInputProps extends Omit<InputProps, 'ref' | 'valu
 }
 
 /** @public */
-export const DatePickerWithInput = ({
-  value,
-  minDate,
-  maxDate,
-  onChange,
-  closeOnSelect,
-  placeholder = 'Date',
-  ...rest
-}: DatePickerWithInputProps) => {
-  const [open, setOpen] = useState(false);
-  const styles = useStyles2(getStyles);
+export const DatePickerWithInput = forwardRef<HTMLInputElement, DatePickerWithInputProps>(
+  ({ value, minDate, maxDate, onChange, closeOnSelect, placeholder = 'Date', ...rest }, ref) => {
+    const [open, setOpen] = useState(false);
+    const styles = useStyles2(getStyles);
 
-  // the order of middleware is important!
-  // see https://floating-ui.com/docs/arrow#order
-  const middleware = [
-    flip({
-      // see https://floating-ui.com/docs/flip#combining-with-shift
-      crossAxis: false,
-      boundary: document.body,
-    }),
-    shift(),
-  ];
+    // the order of middleware is important!
+    // see https://floating-ui.com/docs/arrow#order
+    const middleware = [
+      flip({
+        // see https://floating-ui.com/docs/flip#combining-with-shift
+        crossAxis: false,
+        boundary: document.body,
+      }),
+      shift(),
+    ];
+
+    const { context, refs, floatingStyles } = useFloating({
+      open,
+      placement: 'bottom-start',
+      onOpenChange: setOpen,
+      middleware,
+      whileElementsMounted: autoUpdate,
+      strategy: 'fixed',
+    });
 
-  const { context, refs, floatingStyles } = useFloating({
-    open,
-    placement: 'bottom-start',
-    onOpenChange: setOpen,
-    middleware,
-    whileElementsMounted: autoUpdate,
-    strategy: 'fixed',
-  });
+    const click = useClick(context);
+    const dismiss = useDismiss(context);
+    const { getReferenceProps, getFloatingProps } = useInteractions([dismiss, click]);
 
-  const click = useClick(context);
-  const dismiss = useDismiss(context);
-  const { getReferenceProps, getFloatingProps } = useInteractions([dismiss, click]);
+    useImperativeHandle(ref, () => refs.reference.current as HTMLInputElement, [refs.reference]);
```

Comment:
I know that typing refs is a bit of a mess, but is it possible to do this with some type guards?

Reply:
think you can specify the reference type as a generic to `useFloating`, e.g. `useFloating<HTMLInputElement>`

Reply:
Same issue with `HTMLInputElement`. 
I had to use `ReferenceElement`